Plan/common/src/main/java/com/djrapitops/plan/component/ComponentSvc.java

```diff
@@ -0,0 +1,99 @@
+ *  along with Plan. If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.djrapitops.plan.component;
+
+import com.djrapitops.plan.utilities.logging.ErrorContext;
+import com.djrapitops.plan.utilities.logging.ErrorLogger;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Implementation for {@link ComponentService}.
+ *
+ * @author Vankka
+ */
+@Singleton
+public class ComponentSvc implements ComponentService {
+
+    private final IComponentConverter converter;
+
+    @Inject
+    public ComponentSvc(ErrorLogger errorLogger) {
+        IComponentConverter converter = null;
+        try {
+            Class<?> clazz = Class.forName("com.djrapitops.plan.component.ComponentConverter");
+            converter = (IComponentConverter) clazz.getDeclaredConstructor().newInstance();
+        } catch (ReflectiveOperationException e) {
+            errorLogger.error(e, ErrorContext.builder().related("Could not initialize ComponentConverter").build());
+        }
+        this.converter = converter;
+    }
+
+    @Override
+    public String translateLegacy(String input, char inputCharacter, char outputCharacter) {
+        if (converter != null) {
+            return converter.translate(input, inputCharacter, outputCharacter);
+        }
+        return input.replace(Component.AMPERSAND, Component.SECTION);
+    }
+
+    protected String convert(ComponentImpl component, ComponentOperation operation) {
+        return convert(component, operation, Character.MIN_VALUE);
+    }
+
+    protected String convert(ComponentImpl component, ComponentOperation operation, char inputCharacter) {
+        if (converter != null) {
+            return converter.convert(component, operation, inputCharacter);
+        }
+        return component.getInput();
+    }
+
+    @Override
+    public ComponentImpl fromAutoDetermine(String unknown) {
+        return new ComponentImpl(this, ComponentOperation.AUTO_DETERMINE, unknown);
+    }
+
+    @Override
+    public ComponentImpl fromLegacy(String legacy, char character) {
+        return new ComponentImpl(this, ComponentOperation.LEGACY, legacy, character);
+    }
+
+    @Override
+    public ComponentImpl fromAdventureLegacy(String adventureLegacy, char character) {
+        return new ComponentImpl(this, ComponentOperation.ADVENTURE_LEGACY, adventureLegacy, character);
+    }
+
+    @Override
+    public ComponentImpl fromBungeeLegacy(String bungeeLegacy, char character) {
+        return new ComponentImpl(this, ComponentOperation.BUNGEE_LEGACY, bungeeLegacy, character);
+    }
+
+    @Override
+    public ComponentImpl fromMiniMessage(String miniMessage) {
+        return new ComponentImpl(this, ComponentOperation.MINIMESSAGE, miniMessage);
+    }
+
+    @Override
+    public ComponentImpl fromJson(String json) {
+        return new ComponentImpl(this, ComponentOperation.JSON, json);
+    }
+
+    public void register() {
+        ComponentService.Holder.set(this);
```

Comment:
Should probably throw EnableException if initialization in constructor failed (The message in EnableException is directly displayed to user)